generators/generator-base-blueprint.js

```diff
@@ -95,4 +106,17 @@ module.exports = class extends BaseGenerator {
         }
         return useBlueprints;
     }
+
+    migrateBlueprintConfig() {
+        if (!this.blueprintConfig.get('blueprintConfigOnly')) {
+            // Copy every config to jhipster config
+            this.config.set(this.blueprintConfig.getAll());
```

Comment:
what will happen here, if there is a duplicate value?

Reply:
Don't seems right, but It will be overridden.
Just like getAllJhipsterConfig =/

Reply:
This is the main issue.
For this problem, we can revert last commit and you can use
`const configuration = {...this.getConfig.getAll(), ... this.blueprintConfig.getAll()}`
instead of getAllJhipsterConfig.
Remove the duplicates, but keep the specific ones.

Reply:
In that case, I have touched upon only the things that are related to `Docker` and `k8s` which will need server values that will be present only in the blueprint configuration. But this changes globally for the blueprint. That is the main reason I want to have a proper solution here.

Reply:
I didn't got your point here.
Do you have a blueprint and were it is used?

If it needs the configuration from blueprint, it should get by getAllJhipsterConfig or some way that the blueprint pass the config to the main generator.

Reply:
Can you try testing with Kotlin and VueJS blueprint together?

Reply:
Probably we should let the blueprint migrate their configs.
But create the means to create conflict free configs.

Reply:
If we came up with a good migration strategy, then do it.

Reply:
Removed the auto migration and if the blueprint wants a clean config pass opt.newConfig to the constructor.
WDYT?

Reply:
@mshima this is so weird, what will happen when two server blueprints work together how does the configuration will result? How does blueprint config should look like? I think there should not be any configuration inside blueprint and all of them should live inside the main config. WDYT?

Reply:
> what will happen when two server blueprints work together how does the configuration will result?

Depends on the order of the blueprint, but mainly the config will be duplicated (not guaranteed).

> How does blueprint config should look like?

Like this:
https://github.com/jhipster/generator-jhipster/blob/ae74d36c57f2aa90f680c0fa27fce67e1b11e81e/test/templates/ngx-blueprint/.yo-rc.json

> I think there should not be any configuration inside blueprint and all of them should live inside the main config.

That last migration do that, but it only set a flag on blueprint config to don't migrate anymore.

With current version and the blueprint setting opts.blueprintConfig the new config will alway be generator-jhipster.
Unless the blueprint uses `this.blueprintConfig.set`, then it will be specific config.
But migration should be done by the blueprint.

At my blueprint I create some specific config, so I set to the blueprint config.

Reply:
@sendilkumarn I suggest do the migration copying everything and if the blueprint is "config safe" it sets the opts.blueprintConfig=true.

This will be the inverse of current patch.

Reply:
For example, at:
https://github.com/jhipster/generator-jhipster/blob/ae74d36c57f2aa90f680c0fa27fce67e1b11e81e/test/templates/ngx-blueprint/.yo-rc.json

The config `promptValues` is a conflict.

https://github.com/jhipster/generator-jhipster/blob/ae74d36c57f2aa90f680c0fa27fce67e1b11e81e/test/templates/ngx-blueprint/.yo-rc.json#L2-L5
will be used when:

- The generator is blueprinted
- getAllJhipsterConfig is used.


https://github.com/jhipster/generator-jhipster/blob/ae74d36c57f2aa90f680c0fa27fce67e1b11e81e/test/templates/ngx-blueprint/.yo-rc.json#L27-L30
will be used when:

- App generator (no blueprint support)
- The generator is not blueprinted and not using getAllJhipsterConfig